# HG changeset patch
# User John Donoghue <john.donoghue@ieee.org>
# Date 1390951077 18000
#      Tue Jan 28 18:17:57 2014 -0500
# Branch gui-release
# Node ID 6ebd37faae5f3d701f20acc3fc4d948c6a347f14
# Parent  2ea741d2255446ad0292fb32e3042a178b99895b
Add path separator to end of uigetfile path (Bug #41367)

* libgui/src/dialog.cc
  (FileDialog::accept): Add / to pathname if showing files dialog. Convert path to native separators.

diff --git a/libgui/src/dialog.cc b/libgui/src/dialog.cc
--- a/libgui/src/dialog.cc
+++ b/libgui/src/dialog.cc
@@ -486,18 +486,24 @@ void FileDialog::accept (void)
   string_result = selectedFiles ();
 
   // Matlab expects just the file name, whereas the file dialog gave us
   // pull path names, so fix it.
 
   for (int i = 0; i < string_result.size (); i++)
     string_result[i] = QFileInfo (string_result[i]).fileName ();
 
+  path = directory ().absolutePath ();
 
-  path = directory ().absolutePath ();
+  // if not showing only dirs, add end slash for the path component
+  if (testOption (QFileDialog::ShowDirsOnly)  == false)
+    path = path + "/";
+
+  // convert to native slashes
+  path = QDir::toNativeSeparators (path);
 
   QStringList name_filters = nameFilters ();
   idx = name_filters.indexOf (selectedNameFilter ()) + 1;
 
   // send the selected info
   emit finish_input (string_result, path, idx);
   done (QDialog::Accepted);
 }
